microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/extensions/ItemMetadataProviderFactory.java

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.extensions;
+
+import org.eclipse.lsp4mp.commons.metadata.ItemMetadata;
+
+/**
+ * {@link ItemMetadata} provider factory API.
+ *
+ */
+public interface ItemMetadataProviderFactory {
+
+	/**
+	 * Create an {@link ItemMetadata} provider factory.

```

Comment:
This class is the factory, it makes an ItemMetadataProvider.

Reply:
fixed